Good-Bids/goodbids
# [#179] Auction Details Endpoint

# Summary

This PR adds a new Endpoint (`/wp-json/wp/v2/auction/{ID}/details`) to the WordPress REST API that returns the details about the given Auction ID:

### Endpoint details

* URL: /wp-json/wp/v2/auction/`<id>`/details
* Method: GET
* Authorization: None
* Body: N/A
* Response: JSON object - Data will depend on Auction Status (see below)

| Upcoming      | Live   |  Closed   |
| ----------- | ----------- | ----------- |
| auctionStatus | auctionStatus | auctionStatus | 
| socketUrl | socketUrl | socketUrl |
| accountUrl | accountUrl | accountUrl |
| bidUrl | bidUrl |  |
| startTime | startTime |  | 
| endTime | endTime | endTime | 
|  | totalBids | totalBids | 
|  | totalRaised | totalRaised | 
|  | currentBid |  | 
|  | lastBid | lastBid | 


**All Available Data** - Returned when Auction Status = `live`:
```json
{
  "auctionStatus": "live",
  "socketUrl": "wss://goodbids-node-develop.go-vip.net/_ws/connect",
  "accountUrl": "",
  "bidUrl": "https://goodbids.vipdev.lndo.site/checkout/?add-to-cart=216",
  "startTime": "2023-12-28T12:37:00+00:00",
  "endTime": "2024-01-31T14:01:05+00:00",
  "totalBids": 6,
  "totalRaised": 21,
  "currentBid": 7,
  "lastBid": 6
}
```

## PR Bonuses!

Trying to put a positive spin on shoving some extra code into this PR, but it also adds some code that restricts access to the `/users/` REST API endpoints that only allow Gutenberg access to it. I tested this and wasn't able to see any data at `users` but Gutenberg seemed to function normally.

## Issues

* #179

## Testing Instructions

1. Get the ID of an existing Auction (preferably one with bids/activity.
2. Use Postman (or just visit the URL in your browser) to perform a GET request on the URL:
   * `{{BASE_URL}}`/wp-json/wp/v2/auction/`{ID}`/details
3. You should receive the return data based on the status of the Auction requested.
